dotnet/src/Connectors/Connectors.Memory.SqlServer.Classic/Core/SqlServerClient.cs

```diff
@@ -0,0 +1,473 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.Data.SqlClient;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text.Json;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.SemanticKernel.Connectors.SqlServer.Classic.Core;
+
+/// <summary>
+/// Represents a client for interacting with a SQL Server database for storing semantic memories and embeddings.
+/// </summary>
+[System.Diagnostics.CodeAnalysis.SuppressMessage("Security", "CA2100:Review SQL queries for security vulnerabilities",
+                                                   Justification = "We need to build the full table name using schema and collection, it does not support parameterized passing.")]
+internal sealed class SqlServerClient
+{
+    private readonly string _connectionString;
+    private readonly SqlServerConfig _configuration;
+    private SqlConnection _connection;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SqlServerClient"/> class with the specified connection string and schema.
+    /// </summary>
+    /// <param name="connectionString">The connection string to use for connecting to the SQL Server database.</param>
+    /// <param name="configuration">The configuration to use for the SQL Server database.</param>
+    public SqlServerClient(string connectionString, SqlServerConfig configuration)
+    {
+        this._connectionString = connectionString;
+        this._configuration = configuration;
+
+        this._connection = new SqlConnection(this._connectionString);
+        this._connection.Open();
+    }
+
+    /// <inheritdoc />
+    public async Task CreateTablesAsync(CancellationToken cancellationToken)
```

Comment:
Unfortunately, this is needed.

I think I've got 3 possibilities: 
1. Implement a synchronous method to create needed tables at startup (during the constructor) -> risk to have a long running job during constructor which is not fine
2. Implement the builder extension as this: 
    ```
    return builder.WithMemoryStore<SqlServerMemoryStore>(_ => SqlServerMemoryStore.ConnectAsync(connectionString, config)
                                                                               .ConfigureAwait(false)
                                                                               .GetAwaiter()
                                                                               .GetResult());
                                                                               
    ```  
    This will block the thread during the instanciation of the memoryDb, still not fine I think.
3. Let the user providing and instancing the memory store himself and provide the instance in the builder using the ``public MemoryBuilder WithMemoryStore(IMemoryStore store)`` method

I personally think the .3 is more relevant. More similar to the SQLite implementation. 
@westey-m what is you opinion?

Reply:
IMHO both 1. and 2. should be no-go.

Reply:
@kbeaugrand, have you considered having the same implementation as the Azure SQL Server implementation?
See: https://github.com/microsoft/semantic-kernel/blob/e1ab3455c1c8ce40f434ab512cbb9093ba863c16/dotnet/src/Connectors/Connectors.Memory.SqlServer/SqlServerMemoryStore.cs#L57
It creates a new table for each collection using the collection name.  This means that no tables need to be created at startup time.  A table is only created when CreateCollection is called.

This is also good in that it means we have a consistent approach to modelling collections between the different SQL Server connector implementations.

Reply:
I fixed it by removing this table and using only SK memories tables, by looking into tables with a prefix.